### PR TITLE
New skills viz

### PR DESCRIPTION
**Pull request by india-kerle**
---

# Description

This PR will produce top non-green skills for occupations based on overall greenness. the script needs to be updated to:
- use soc 2020 ext 
- the occ_aggregated dataset
- tfidfs by 'greenness score'

Fixes # (issue)

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

**Comment by india-kerle** [comment]
I've re-run the `aggregate_by_sic.py` and `aggregate_by_soc.py` scripts and updated the `ojo_analysis.yaml` file to reflect the most recent versions of the datasets.

**Comment by lizgzil** [comment]
@india-kerle my commit fixed that issue.
Have no idea why `filter_large_occs` didn't work, but I also added a manual step for it.

I have rerun the aggregate datasets.

I didn't rerun the notebooks/vizs, but I think we need to update them later down the line with Laurens suggestions etc, so can just leave for then?



I think you'll need to rerun the skill viz with the tweak:
```
full_skill_mapping = pg.load_full_skill_mapping(analysis_config)

  skills_df = pg.create_skill_df(
      green_skills_outputs, full_skill_mapping
  )
  ```

File: dap_prinz_green_jobs/config/ojo_analysis.yaml
Comment (lizgzil) [inline dap_prinz_green_jobs/config/ojo_analysis.yaml]: should be "large"

**Comment by lizgzil** [comment]
fyi @india-kerle I noticed this hadn't been rebased with the fix to data_type="large" (from "mixed") in the ojo_analysis config, which meant the latest dataset for the regional aggregation (20231220) had been rerun on the mixed sample. I've rerun this today with datatype="large".

Think we need a double checking time when everything is merged where we check over the aggregation scripts, rerun them all, and rerun all the viz notebooks.

**Comment by india-kerle** [comment]
@lizgzil wow thanks so much for this - I've kept this branch has been open for far too long given how actively we're updating analysis graphs. I'll prioritise getting this in a state for review by EOD or tomorrow so we can just merge ASAP. Also agreed on double checking time given this.

**Comment by india-kerle** [comment]
@lizgzil the network approach seems to be more interesting than the top tf-idf approach - i've created two graphs - 1 is just to explore the network of green-non-green skills in highly green occupations while the second is the bar chart of top non-green skills based on degree centrality in the network. it might be nice to filter by occupation etc. but i think let's just merge this and if there's any other altair graphs we want to make, we can do so in a new PR.

**Review by lizgzil — APPROVED** [review]
looks good! love the new plots 😸

**Pull request merged**
